content/posts/how-to-create-custom-tables/index.md

```diff
@@ -0,0 +1,233 @@
+
+```python
+# highlight the column we are sorting by
+# using a rectangle patch
+rect = patches.Rectangle(
+	(1.5, -.5),  # bottom left starting position (x,y)
+	.65,  # width
+	10,  # height
+	ec='none',
+	fc='grey',
+	alpha=.2,
+	zorder=-1
+)
+ax.add_patch(rect)
+```
+
+![Highlight column](5_highlight_column.png)
+
+We're almost there. The magic piece is `ax.axis(‘off’)`. This hides the axis, axis ticks, labels and everything “attached” to the axes, which means our table now looks like a clean table!
+
+```python
+ax.axis('off')
+```
+
+![Hide axis](6_hide_axis.png)
+
+Adding a title is also straightforward.
+
+```python
+ax.set_title(
+	'A title for our table!',
+	loc='left',
+	fontsize=18,
+	weight='bold'
+)
+```
+
+# Bonus: Adding special columns
+
+Finally, if you wish to add images, sparklines, or other custom shapes and patterns then we can do this too.
+
+To achieve this we will create new floating axes using `fig.add_axes()` to create a new set of floating axes based on the figure coordinates (this is different to our axes coordinate system!).
+
+Remember that figure coordinates by default are between 0 and 1. [0,0] is the bottom left corner of the entire figure. If you’re unfamiliar with the differences between a figure and axes then check out [Matplotlib's Anatomy of a Figure](https://matplotlib.org/stable/gallery/showcase/anatomy.html) for further details.
+
+```python
+newaxes = []
+for row in range(rows):
+    # offset each new axes by a set amount depending on the row
+    # this is probably the most fiddly aspect (TODO: some neater way to automate this)
+    newaxes.append(
+        fig.add_axes([.75, .725 - (row*.063), .12, .06])
+    )
+```
+
+You can see below what these *floating* axes will look like (I say floating because they’re on top of our main axis object). The only tricky thing is figuring out the xy (figure) coordinates for these.
+
+These *floating* axes behave like any other Matplotlib axes. Therefore, we have access to the same methods such as ax.bar(), ax.plot(), patches, etc. Importantly, each axis has its own independent coordinate system. We can format them as we wish.
+
+![Floating axes](7_floating_axes.png)
+
+```python
+# plot dummy data as a sparkline for illustration purposes
+# you can plot _anything_ here, images, patches, etc.
+newaxes[0].plot([0, 1, 2, 3], [1, 2, 0, 2], c='black')
+newaxes[0].set_ylim(-1, 3)
+
+# once again, the key is to hide the axis!
+newaxes[0].axis('off')
+```
+
+![Sparklines](8_sparklines.png)
+
+That’s it, custom tables in Matplotlib. I did promise very simple code and an ultra-flexible design in terms of what you want / need. You can adjust sizes, colors and pretty much anything with this approach and all you need is simply a loop that plots text in a structured and organized manner. I hope you found it useful.

```

Comment:
Can you add a link to the full code / jupyter notebook for the reader to download?